R/R/run.R

```diff
@@ -0,0 +1,166 @@
+#' Run metaflow
+#'
+#' @description
+#' `run()` passes all command line arguments to metaflow.
+#' These are captured whether running from interactive session or via `Rscript`
+#'
+#'
+#' @param flow metaflow object
+#' @param ... passed command line arguments
+#' @details Command line arguments:
+#' * package_suffixes: any file suffixes to include in the run
+#'     * ex: c('.csv', '.R', '.py')
+#' * datastore: 'local' (default) or 's3' 
+#' * metadata:  'local' (default) or 'service'
+#' * batch: request flow to run on batch (default FALSE)
+#' * resume: resume flow from last failed step
+#'     * logical (default FALSE)
+#' * with: any flow level decorators to include in the run
+#'     * ex: c('retry', 'batch', 'catch')
+#' * max_workers: limits the number of tasks run in parallel
+#' * max_num_splits: maximum number of parallel splits allowed
+#' * other_args: escape hatch to provide args not covered above
+#' * key=value: any parameters specified as part of the flow
+#' @section Usage:
+#' \preformatted{
+#' run(flow, batch = TRUE, with = c("retry", "catch"), max_workers = 16, max_num_splits = 200)
+#' run(flow, alpha = 0.01)
+#' }
+#' @export
+run <- function(flow = NULL, ...) {
+  flow_file = tempfile(flow$get_name(), tmpdir=".", fileext=".RDS")
+  tryCatch({
+    saveRDS(flow, flow_file)
+  }, error=function(e){
+    stop(sprintf("Cannot create temporary RDS file %s", flow_file))
+  })
+  
+  run_cmd <- run_cmd(flow_file = flow_file, ...)
+  message(paste0("Flow cli:\n", run_cmd))
+  status_code <- system(run_cmd)
+  invisible(file.remove(flow_file))
+  return(status_code)
+}
+
+run_cmd <- function(flow_file, ...) {
+  flags <- flags(...)
+  run_path <- system.file("run.R", package = "metaflow")
+  if ("metadata" %in% names(flags)) {
+    metadata <- paste0("--metadata=",flags$metadata)
+  } else {
+    metadata <- "--metadata=local"
+  }
+  if ("datastore" %in% names(flags)) {
+    datastore <- paste0("--datastore=",flags$datastore)
+  } else {
+    datastore <- "--datastore=local"
+  }
+  if ("package_suffixes" %in% names(flags)) {
+    package_suffixes <- paste0("--package-suffixes=", paste(union(flags$package_suffixes, c(".R", ".py")), collapse = ","))
+  } else {
+    package_suffixes <- "--package-suffixes=.R,.py"
+  }
+  if ("resume" %in% names(flags)) {
+    if (is.logical(flags$resume)) {
+      if (flags$resume) {
+        run <- "resume"
+      }
+    } else {
+      run <- paste0("resume", " ", flags$resume)
+    }
+    if ("origin_run_id" %in% names(flags)) {
+      run <- paste0(run, " --origin-run-id=", flags$origin_run_id)
+    }
+  } else {
+    run <- "run"
+  }
+  
+  if ("batch" %in% names(flags)) {
+    if (is.logical(flags$batch)) {
+      if (flags$batch) {
+        batch <- "--with batch"
+        metadata <- "--metadata=service"
```

Comment:
`metadata=service` is not a strict requirement here. We can avoid setting `metadata` and `datastore` and just rely on the defaults from metaflow-py.

Reply:
Makes sense. Will make the change.